x/liquidity/client/cli/tx.go

```diff
@@ -0,0 +1,331 @@
+			if err != nil {
+				return err
+			}
+			withdrawer := clientCtx.GetFromAddress()
+
+			// Get pool type index
+			poolId, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("pool-id %s not a valid uint, input a valid unsigned 32-bit integer for pool-id", args[0])
+			}
+
+			// Get pool coin of the target pool
+			poolCoin, err := sdk.ParseCoinNormalized(args[1])
+			if err != nil {
+				return err
+			}
+
+			err = poolCoin.Validate()
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgWithdrawWithinBatch(withdrawer, poolId, poolCoin)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// Swap offer coin with demand coin from the specified liquidity pool with the given order price.
+func NewSwapWithinBatchCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "swap [pool-id] [swap-type] [offer-coin] [demand-coin-denom] [order-price] [swap-fee-rate]",
+		Args:  cobra.ExactArgs(6),
+		Short: "Swap offer coin with demand coin from the liquidity pool with the given order price",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Swap offer coin with demand coin from the liquidity pool with the given order price.
+
+This swap request is not processed immediately since it is accumulated in the liquidity pool batch.
+All requests in a batch are treated equally and executed at the same swap price.
+The order of swap requests is ignored since the universal swap price is calculated in every batch to prevent front running.
+
+The requested swap is executed with a swap price that is calculated from the given swap price function of the pool, the other swap requests, and the liquidity pool coin reserve status.
+Swap orders are executed only when the execution swap price is equal to or greater than the submitted order price of the swap order.
+
+Example:
+$ %s liquidityd tx liquidity swap 1 1 50000000uusd uatom 0.019 0.003 --from mykey
+
+For this example, imagine that an existing liquidity pool has with 1000000000uatom and 50000000000uusd.
+This example request swaps 50000000uusd for at least 950000uatom with the order price of 0.019 and swap fee rate of 0.003.
+A sufficient balance of half of the swap-fee-rate of the offer coin is required to reserve the offer coin fee.
+
+The order price is the exchange ratio of X/Y, where X is the amount of the first coin and Y is the amount of the second coin when their denoms are sorted alphabetically.
+Increasing order price reduces the possibility for your request to be processed and results in buying uatom at a lower price than the pool price.
+
+For explicit calculations, The swap fee rate must be the value that set as liquidity parameter in the current network.
+The only supported swap-type is 1. For the detailed swap algorithm, see https://github.com/tendermint/liquidity
```

Comment:
The only supported swap-type is 1. For swap algorithm details, see the [light paper](https://github.com/tendermint/liquidity/blob/develop/doc/LiquidityModuleLightPaper_EN.pdf).